GUI/QtQuick/SideshiftPlugin.cpp

```diff
@@ -181,17 +266,22 @@ void SideshiftPlugin::init()
       const auto& response = reply->response;
       try {
          const auto& msg = json::parse(response);
+         QList<Currency> currencies;
          for (const auto& coin : msg) {
-            const auto& currency = coin["coin"].get<std::string>();
-            auto& networks = networksByCur_[currency];
             for (const auto& network : coin["networks"]) {
-               networks.append(QString::fromStdString(network.get<std::string>()));
+               currencies.append({
+                  QString::fromStdString(coin["name"]),
+                  QString::fromStdString(coin["coin"]),
+                  QString::fromStdString(network.get<std::string>())
+               });
             }
-            inputCurrencies_.append(QString::fromStdString(currency));
          }
+
+         inputListModel_->reset(currencies);
+         outputListModel_->reset({ {tr("Bitcoin"), tr("BTC"), tr("bitcoin")} });
```

Comment:
This should be done in `invokeMethod` wrapper, as the lambda is executed in the curl/network thread.